client/src/Components/EditPostPopup.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import Modal from 'react-bootstrap/Modal'
+import Button from 'react-bootstrap/Button'
+import { useState } from 'react';
+import Input from './Input';
+import Textarea from './Textarea';
+
+export default function EditPostPopup({show , onClose, post, onUpdate}) {
+    
+  const [editedPost, setEditedPost] = useState(post)
+  
+  function handleChange (event) {
+    const { value, name } = event.target;
+    setEditedPost((prevValue) => ({ ...prevValue, [name]: value }));
+  }
+
+  const editThePost= async (post) => {
+    try {
+      const body = { ...post };
+
+      const response = await fetch(`http://localhost:1212/${post.id}`, {
+        method: "PUT",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(body)
```

Comment:
```suggestion
        body: JSON.stringify(post)
```

You don't need the `const body` variable in this scope. You only need to do this shallow copy whenever you need to make edits to the object.